e2e/test/scenarios/onboarding/search/search.cy.spec.js

```diff
@@ -1092,7 +1099,384 @@ describeWithSnowplow("scenarios > search", () => {
     cy.visit("/");
     expectGoodSnowplowEvents(PAGE_VIEW_EVENT);
     getSearchBar().type("Orders").blur();
-    expectGoodSnowplowEvents(PAGE_VIEW_EVENT + 1); // new_search_query
+    expectGoodSnowplowEvents(PAGE_VIEW_EVENT + NEW_SEARCH_QUERY_EVENT);
+  });
+
+  describe("should send snowplow events for each filter when it is applied and removed", () => {
+    describe("no filters", () => {
+      it("should send a snowplow event when a search with no filters is accessed from the URL", () => {
+        cy.visit("/search?q=orders");
+        expectGoodSnowplowEvents(
+          PAGE_VIEW_EVENT +
+            NEW_SEARCH_QUERY_EVENT +
+            SEARCH_RESULT_FILTERED_EVENT,
+        );
+      });
+    });
+
+    describe("type filter", () => {
+      it("should send a snowplow event when a search filter is used in the URL", () => {
+        cy.visit("/search?q=orders&type=card");
+        // TODO: this should return +2 events - new_search_query event isn't firing
```

Comment:
For some reason the `new_search_query` event isn't firing when we use the `type` filter - I'll need to check with the BE on this one since they fire the `new_search_query` event

Reply:
@qnkhuat would you know what's happening here? I've tried the other filters locally and with this test env and the type filter is the only filter that doesn't trigger `new_search_query`

Reply:
yea in BE we explictly track when there are no type filters [here](https://github.com/metabase/metabase/blob/3f2afbf9fe2e7eb2f152a50d89e5f15304cb0776/src/metabase/api/search.clj#L518).

It was from before we have this type filter and the assumption is that we only track the search from the global search view. search from other components like selecting cards to add to a dashboard shouldn't get tracked.

but now with the new filters, I think we should be smarter. wdyt if we add a new flag `is_global_search`, and FE fill it with `true` on the global search view so we can track it correctly?

Reply:
Maybe we can just remove that extra check for the `models` parameter for now? We'll need to check in with PDs if we want to make more changes

Reply:
Hm... the more I think about it, the more I'm convinced that we should probably have an enum or a flag to tell us where we are searching. @cdeweyx what do you think about having a `location` flag with `global-search` or `search-bar` or something? Since the `new_search_query` event fires when a user is using the search bar, and we probably want to see how often a user is using the filters the global search (against using no filters in the global search), I think it would help us differentiate the events a little more easily

Reply:
Yeah I think that's a good idea. For clarity, we're distinguishing between searching globally using the search bar in the top nav vs. searching with filters? If so then I'd say something like a `location` property with `global_search` or `filtered_search` would do the trick.